App/App.cpp

```diff
@@ -1,28 +1,48 @@
-#include <stdio.h>
-#include <climits>
-#include "App.h"
-#include "SgxException.h"
-#include "EnclaveManager.h"
-
-long affineTransformation(unsigned long x, long targetRangeMin, long targetRangeMax, unsigned long sourceRangeMin, unsigned long sourceRangeMax) {
-	return ((double)x - (double)sourceRangeMin)*((double)targetRangeMax - (double)targetRangeMin) / ((double)sourceRangeMax - (double)sourceRangeMin) + (double)targetRangeMin;
-}
-
-int generateRandom(long min, long max, long *result) {
-	try {
-		EnclaveManager enclave;
-		unsigned long generated = enclave.generateRandom();
-
-		*result = affineTransformation(generated, min, max, 0, ULONG_MAX);
-		printf("Transforming random value from SGX [0, %lu] to [%ld, %ld]: %lu -> %ld \n", ULONG_MAX, min, max, generated, *result);
-	} catch (SgxException& e) {
-		printf("%s\n", e.what());
-		return -1;
-	}
-
-	return 0;
-}
-
-int main() {
-	return 0;
-}
+#include <stdio.h>
+#include <climits>
+#include "App.h"
+#include "SgxException.h"
+#include "EnclaveManager.h"
+
+long affineTransformation(unsigned long x, long targetRangeMin, long targetRangeMax, unsigned long sourceRangeMin, unsigned long sourceRangeMax) {
+	return ((double)x - (double)sourceRangeMin)*((double)targetRangeMax - (double)targetRangeMin) / ((double)sourceRangeMax - (double)sourceRangeMin) + (double)targetRangeMin;
+}
+
+int generateRandom(long min, long max, long *result) {
+	try {
+		EnclaveManager enclave;
+		unsigned long generated = enclave.generateRandom();
+
+		*result = affineTransformation(generated, min, max, 0, ULONG_MAX);
+		printf("Transforming random value from SGX [0, %lu] to [%ld, %ld]: %lu -> %ld \n", ULONG_MAX, min, max, generated, *result);
+	} catch (SgxException& e) {
+		printf("%s\n", e.what());
+		return -1;
+	}
+
+	return 0;
+}
+
+int initRemoteAttestation() {
+    try {
+		EnclaveManager enclave;
```

Comment:
You're creating `enclave` here and then initializing remote attestation. But after that when function returns, the enclave will be destroyed. If it's that what you want then I think the name of the `initRemoteAttestation` function should be changed because `init` suggests that some work (after initialization) is yet to be done.

Reply:
Yes, this name should be changed. It doesn't just init RA - it performs it.